web/src/app/alerts/components/AlertDetailLogs.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import p from 'prop-types'
+import Divider from '@material-ui/core/Divider'
+import List from '@material-ui/core/List'
+import ListItem from '@material-ui/core/ListItem'
+import ListItemText from '@material-ui/core/ListItemText'
+import Button from '@material-ui/core/Button'
+import moment from 'moment'
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+
+const LIMIT = 149
+
+const query = gql`
+  query getAlert($id: Int!, $input: AlertRecentEventsOptions) {
+    alert(id: $id) {
+      data: recentEvents(input: $input) {
+        nodes {
+          timestamp
+          message
+        }
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+      }
+    }
+  }
+`
+export default function AlertDetailLogs(props) {
+  let logs = []
+  const { data, startPolling, stopPolling, fetchMore } = useQuery(query, {
+    variables: { id: props.alertID, input: { after: '', limit: 35 } },
+  })
+  if (data) {
+    if (data.alert.data.nodes.length === 0) {
+      return (
+        <ListItem>
+          <ListItemText primary='No events.' />
+        </ListItem>
+      )
+    }
+    if (data.alert.data.nodes.length <= 35) {
+      startPolling(2500)
+    } else stopPolling()
+
+    if (props.showExactTimes) {
```

Comment:
I think we can simplify this all by breaking it up so we don't have so many nested `if` statements. Something like this as an idea:

```js
const doFetchMore = () => { /* fetchMore logic here */}
const renderList = (items, loadMore) => (
    <List data-cy='alert-logs' style={{ padding: 0 }}>
        {items}
        {loadMore && <Button onClick={doFetchMore}>Load More</Button>}
    </List>
)
const renderTime = ts => exactTimes ? luxon_magic1(ts) : luxon_magic2(ts)

if (error) return renderList( <ListItem><ListItemText primary={error.message} /></ListItem>) 
if (loading && !data) return renderList( <ListItem><ListItemText primary='Loading...' /></ListItem>)
if (data.nodes.length===0) return renderList( <ListItem><ListItemText primary='No events.' /></ListItem>)

return renderList(nodes.map((log,index)=>(
  <div key={index}> ... renderTime(log.timestamp) ... </div>
)), data.hasNextPage)
```